pkg/experiment/metrics/rules.go

```diff
@@ -48,5 +87,101 @@ func recordingRulesFromTenant(tenant string) []*RecordingRule {
 			},
 			keepLabels: []string{},
 		},
+		{
+			profileType: "process_cpu:cpu:nanoseconds:cpu:nanoseconds",
+			metricName:  "pyroscope_exported_metrics_mimir_dev_10_ingester",
+			matchers: []*labels.Matcher{
+				{
+					Type:  labels.MatchEqual,
+					Name:  "service_name",
+					Value: "mimir-dev-10/ingester",
+				},
+			},
+			keepLabels: []string{"controller_revision_hash"},
+		},
+		{
+			profileType: "process_cpu:cpu:nanoseconds:cpu:nanoseconds",
+			metricName:  "pyroscope_exported_metrics_mimir_dev_10_ingester_with_span_names",
+			matchers: []*labels.Matcher{
+				{
+					Type:  labels.MatchEqual,
+					Name:  "service_name",
+					Value: "mimir-dev-10/ingester",
+				},
+			},
+			keepLabels: []string{"controller_revision_hash", "span_name"},
+		},
+	}
+}
+
+func SettingsClient() settingsv1connect.SettingsServiceClient {
+	// TODO: refactor this. Very rudimentary, only intended for experimental
+	httpClient := util.InstrumentedDefaultHTTPClient()
+	opts := connectapi.DefaultClientOptions()
+	opts = append(opts, connect.WithInterceptors(tenant.NewAuthInterceptor(true)))
+	return settingsv1connect.NewSettingsServiceClient(
+		httpClient,
+		"http://fire-tenant-settings-headless.fire-dev-001.svc.cluster.local:4100", // TODO: get it from config (set it in deployment tools) use "http://localhost:4040" for local
+		opts...,
+	)
+}
+
+func parseRule(rule *settingsv1.Setting) *RecordingRule {
+	var parsed RecordingRuleSetting
+	err := json.Unmarshal([]byte(rule.Value), &parsed)
+	if err != nil {
+		// TODO
+		fmt.Println("Error parsing JSON:", err)
+	}
+
+	return &RecordingRule{
+		profileType: parsed.ProfileType,
+		metricName:  "pyroscope_exported_metrics_" + parsed.Name, // TODO sanitize
+		matchers:    parseMatchers(parsed.Matcher, parsed.ServiceName),
+		keepLabels:  parsed.Labels, // [] != All
+	}
+}
+
+func parseMatchers(matchersString string, serviceName string) []*labels.Matcher {
```

Comment:
`github.com/prometheus/prometheus/promql/parser.ParseMetricSelector` should be used